backends-clickhouse/src/main/scala/org/apache/gluten/backendsapi/clickhouse/CHBackend.scala

```diff
@@ -326,6 +326,8 @@ object CHBackendSettings extends BackendSettingsApi with Logging {
 
   override def supportStructType(): Boolean = true
 
+  override def structFieldToLowerCase(): Boolean = false
```

Comment:
@taiyang-li, curious whether this issue exists in ch backend. If the fix is applicable to both backends, we can remove this separated handling.

Reply:
ch backend seems to be different. I didn’t add `structFieldToLowerCase` at first, but ck ci reported an error.

Reply:
@PHILO-HE  it seems that CH haven't met this kind of issues. Let's keep it to be false for CH backend.

Reply:
@taiyang-li, thanks for your verification!